### PR TITLE
[OPENY-131] Demo content cleaner - migrations data and configs

### PR DESCRIPTION
**Pull request by anprok**
At this moment only extended or standard installation can be checked. Complete installation is impossible at this moment so Fresh OpenY installation build does not work correctly
## Steps for review
- [x] select Installation process build and use extended preset during installation
- [x] open installed site with extended demo content and log in as admin
- [x] make sure that content and main menu displayed
- [x] visit admin/modules/
- [x] type "demo" in modules filter field
- [x] make sure no demo modules installed
- [x] visit admin/structure/migrate
- [x] make sure that no demo migrations in list (only default one).
- [x] visit admin/config/development/configuration/single/export
- [x] select in _Configuration type_ drop-down Migration.
- [x] make sure that no migration configs listed  in _Configuration name_ dropdown field.

## General checks
- [x] All coding styles are fulfilled and there are no any issues reported by CodeSniffer CI. <br/>CI code sniffer errors
- [x] All tests are running and there are no failed tests reported by CI. <br/>Behat test results
- [x] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [x] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [x] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [x] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [x] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

**Comment by gundevel** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

File: modules/custom/openy_migrate/src/Importer.php [hunk 1]
```diff
@@ -69,4 +80,26 @@ public function importByTag($migration_tag) {
...
+    $dependencies = $this->configManager->findConfigEntityDependentsAsEntities('module', (array) $module_name);
+    // Create array of dependent migrations for module.
+    // That module should be listed in dependencies of migration config.
+    foreach ($dependencies as $dependency) {
+      /** @var \Drupal\migrate_plus\Entity\Migration $dependency */
+      if ($dependency->getEntityTypeId() == 'migration') {
+        $migration_list[] = $dependency->get('id');
+      }
+    }
+    $migrations = $this->migrationManager->createInstances($migration_list);
```
Comment (podarok) [inline modules/custom/openy_migrate/src/Importer.php]: what happens when ```$migration_list == (array)NULL``` ?

File: modules/openy_features/openy_demo_content/modules/openy_demo_nprogram/openy_demo_nprogram.install [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * @file
+ * OpenY Demo Node Program install file.
+ */
+
+/**
+ * Implements hook_uninstall().
+ */
+function openy_demo_nprogram_uninstall() {
```
Comment (podarok) [inline modules/openy_features/openy_demo_content/modules/openy_demo_nprogram/openy_demo_nprogram.install]: these are equal across all modules, could we extend our Uninstall Service with automated hook_uninstall() handling too? @hamrant @anprok

**Review by podarok — CHANGES_REQUESTED** [review]
Small change about NULL array and question

File: modules/openy_features/openy_demo_content/modules/openy_demo_nprogram/openy_demo_nprogram.install (hunk 2 above)
Comment (hamrant) [reply modules/openy_features/openy_demo_content/modules/openy_demo_nprogram/openy_demo_nprogram.install]: @anprok `\Drupal::service('openy.modules_manager')->postUninstall($module_name);` already executed after module uninstall.

I think we need to move `destroyMigrationData` function to `openy.modules_manager` service. And provide logic here for detecting that uninstalled module related to demo content and in this case run `destroyMigrationData` function.

As result - you can remove all this `hook_uninstalls` form PR.

cc @podarok

File: modules/custom/openy_migrate/src/Importer.php [hunk 3]
```diff
@@ -69,4 +80,26 @@ public function importByTag($migration_tag) {
     }
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function destroyMigrationData($module_name) {
+    // Get config entities that are dependent on module.
+    $dependencies = $this->configManager->findConfigEntityDependentsAsEntities('module', (array) $module_name);
```
Comment (ddrozdik) [inline modules/custom/openy_migrate/src/Importer.php]: $dependencies variable can be empty. Let's add a condition to make sure you have something there

File: modules/openy_features/openy_demo_content/modules/openy_demo_nprogram/openy_demo_nprogram.install (hunk 2 above)
Comment (ddrozdik) [reply modules/openy_features/openy_demo_content/modules/openy_demo_nprogram/openy_demo_nprogram.install]: I think we can run `destroyMigrationData` every time, this method just should check is there anything to destroy or not. It should not be limited to the demo content only,if other modules will have migration configs this method should handle them as a well

**Review by ddrozdik — CHANGES_REQUESTED** [review]

File: modules/openy_features/openy_demo_content/modules/openy_demo_nevent/config/install/migrate_plus.migration.file.yml [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
     - file
+  enforced:
+    module:
+      - openy_demo_nevent
 id: openy_demo_nevent_file
 migration_tags:
   - 'OpenY Demo Event'
 migration_group: openy_demo_nevent
-label: 'Files'
+label: 'Import demo eveny files'
```
Comment (ddrozdik) [inline modules/openy_features/openy_demo_content/modules/openy_demo_nevent/config/install/migrate_plus.migration.file.yml]: *eveny* -> event

File: modules/custom/openy_system/openy_system.module [hunk 5]
```diff
@@ -24,5 +24,9 @@ function openy_system_modules_uninstalled(array $modules) {
       // Run post uninstall only for openy modules.
       $modules_manager->postUninstall($module);
     }
+    if (strpos($module, 'openy_demo_') !== FALSE) {
```
Comment (ddrozdik) [inline modules/custom/openy_system/openy_system.module]: this condition limits execution only to the demo modules, but there are other modules with migrations as well, and they leave migration tables after uninstallation. 
See openy_daxko2, openy_gxp modules

**Review by ddrozdik — CHANGES_REQUESTED** [review]

**Review by ddrozdik — APPROVED** [review]

**Comment by ddrozdik** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build858 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build858 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build858 |
| Installation process | http://install.openy.cibox.tools/build858/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/858/

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build858/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build858/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build858/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build858/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build858/blogWCAG2AAhtmlcs.txt

**Comment by ddrozdik** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build864 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build864 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build864 |
| Installation process | http://install.openy.cibox.tools/build864/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/864/

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build864/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build864/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build864/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build864/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build864/blogWCAG2AAhtmlcs.txt

**Comment by ddrozdik** [comment]
@anprok 
migrations test 2018-08-31 22-55-49

**Review by ddrozdik — APPROVED** [review]
OK

**Comment by ddrozdik** [comment]
retest this please

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build871/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build871/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build871/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build871/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build871/blogWCAG2AAhtmlcs.txt

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build871 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build871 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build871 |
| Installation process | http://install.openy.cibox.tools/build871/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/871/

**Comment by ddrozdik** [comment]
retest this please

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build873/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build873/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build873/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build873/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build873/blogWCAG2AAhtmlcs.txt

**Comment by gundevel** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build873 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build873 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build873 |
| Installation process | http://install.openy.cibox.tools/build873/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/873/

**Comment by ddrozdik** [comment]
standard version - OK

**Comment by ddrozdik** [comment]
retest this please

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build874 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build874 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build874 |
| Installation process | http://install.openy.cibox.tools/build874/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/874/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build874/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build874/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build874/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build874/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build874/blogWCAG2AAhtmlcs.txt

**Comment by ddrozdik** [comment]
extended version - OK

**Pull request merged**